lib/biokbase/catalog/registrar.py

```diff
@@ -671,7 +671,10 @@ def run_docker_container(self, dockerclient, image_name, token,
                     self.log(line)
                     line = ""
                 else:
-                    line += char
+                    try:
+                        line += char
+                    except TypeError:
+                        line += char.decode('utf-8')
```

Comment:
I know this was here before you messed with it but building up a string one char at a time is really inefficient as strings are immutable. It would be better to store the chars in a list, then join them when new lines come along. Also wondering what cases trigger dockerclient.logs to return bytes rather than strings as I've not encountered this issue in CI.